modules/apps/frontend-js/frontend-js-react-web/npmscripts.config.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+const preset = require('@liferay/npm-scripts/src/presets/standard');
+
+module.exports = {
+	federation: true,
```

Comment:
I'm not thrilled about this... maybe the CLI and updating SF is a better choice after all.

Having to create this whole file to do `federation: true` sounds like a bit of overkill.

Reply:
Couldn't we detect it based on something else that's already there?

Reply:
The problems when going with the switch are that:

1. The npm-scripts are not really prepared to handle command line switches (I mean internally, in the code). It was a pain to propagate it from the top to the lower levels.
2. We need to modify the SF process.
3. We don't get any benefit for this extra work (1+2) because this is temporary until we migrate everything to webpack 5, which should happen soon or not happen at all. If we migrate everything we will remove the `federation` flag from everywhere. And if we don't achieve it, the same (I'm unsure if `we` or `others` would remove the flag, but someone would have to do it anyway...).

Regarding the detection: it's impossible because, by definition, there's nothing in the projects that mark them as webpack 5 or bundler 2. In fact, what we are doing is to have some projects built with both tools, so they would match the criteria for both types.